encoder/vaapiencoder_h264.cpp

```diff
@@ -1275,6 +1299,82 @@ YamiStatus VaapiEncoderH264::getCodecConfig(VideoEncOutputBuffer* outBuffer)
     return m_headers->getCodecConfig(outBuffer);
 }
 
+void VaapiEncoderH264::fill(
+    VAEncMiscParameterTemporalLayerStructure* layerParam) const
+{
+    layerParam->number_of_layers = m_temporalLayerNum;
+    layerParam->periodicity = H264_MIN_TEMPORAL_GOP;
+
+    for (uint32_t i = 0; i < layerParam->periodicity; i++)
+        layerParam->layer_id[i] = getTemporalId(m_temporalLayerNum, i + 1);
+}
+
+void VaapiEncoderH264::fill(VAEncMiscParameterRateControl* rateControl,
+                            uint32_t temporalId) const
+{
+    rateControl->bits_per_second
+        = m_videoParamCommon.rcParams.layerBitRate[temporalId];
+    rateControl->initial_qp = m_videoParamCommon.rcParams.initQP;
+    rateControl->min_qp = m_videoParamCommon.rcParams.minQP;
+    /*FIXME: where to find max_qp */
+    rateControl->window_size = m_videoParamCommon.rcParams.windowSize;
+    rateControl->target_percentage
+        = m_videoParamCommon.rcParams.targetPercentage;
+    rateControl->rc_flags.bits.disable_frame_skip
+        = m_videoParamCommon.rcParams.disableFrameSkip;
+    rateControl->rc_flags.bits.disable_bit_stuffing
+        = m_videoParamCommon.rcParams.disableBitsStuffing;
+    rateControl->rc_flags.bits.temporal_id = temporalId;
+}
+
+void VaapiEncoderH264::fill(VAEncMiscParameterFrameRate* frameRate,
+                            uint32_t temporalId) const
+{
+    uint32_t expTemId = (1 << (m_temporalLayerNum - 1 - temporalId));
+    if (fps() % expTemId == 0)
+        frameRate->framerate = fps() / expTemId;
+    else
+        frameRate->framerate = (expTemId << 16 | fps());
+
+    frameRate->framerate_flags.bits.temporal_id = temporalId;
+}
+
+/* Generates additional control parameters */
+bool VaapiEncoderH264::ensureMiscParams(VaapiEncPicture* picture)
+{
+    VAEncMiscParameterHRD* hrd = NULL;
+    if (!picture->newMisc(VAEncMiscParameterTypeHRD, hrd))
+        return false;
+    if (hrd)
+        VaapiEncoderBase::fill(hrd);
+    VideoRateControl mode = rateControlMode();
+    if (mode == RATE_CONTROL_CBR || mode == RATE_CONTROL_VBR) {
+
+        VAEncMiscParameterTemporalLayerStructure* layerParam = NULL;
```

Comment:
Yeah, it is a good suggestion to make sure h264 can be build-able for libva 0.36 version.  I'll update it.